application/src/main/java/org/togetherjava/tjbot/commands/mathcommands/TeXCommand.java

```diff
@@ -57,6 +59,19 @@ public void onSlashCommand(@NotNull final SlashCommandEvent event) {
         String userID = (Objects.requireNonNull(event.getMember()).getId());
         TeXFormula formula;
         try {
+            String regex = "(\\$\\S+\\$)|([a-zA-Z!., ]+)";
+            Pattern pattern = Pattern.compile(regex);
```

Comment:
Right now you constantly recompile this pattern. Instead, put it into a field (constant):
```java
private static final Pattern GIVE_ME_A_NICE_NAME = Pattern.compile("...");
```